### ISSUE TITLE
Frozen disk leads to unavailability

### ISSUE THREAD

**Issue by bdarnell** [cockroachdb/cockroach#32736]
If the disk simply stops responding to writes on a single node of the cluster, severe availability problems can result. (I'm not sure yet what exactly causes such a thing, but I've seen it happen twice today with EBS volumes). If the writes *fail* the process will probably crash, but we don't have any sort of timeouts in place to detect if they are simply taking a long time. 

There are a couple of ways that this problem on a single node can result in larger problems:
- If the logging disk is unresponsive, then any thread that tries to log can pile up waiting on the logging mutex. 
- The node with the bad disk can probably maintain its liveness record, since the other two nodes (assuming a three-node cluster) are enough to commit entries to the liveness range. Therefore it can hold epoch-based leases indefinitely and prevent other nodes from being able to serve those ranges.

We should detect when it's taking too long to write to disk and either kill the process or at least fail health checks and relinquish its leases. The `server.clock.persist_upper_bound_interval` setting might be a good way to enforce that since it would be one place that does local disk writes on a regular interval.

**Comment by knz** [comment]
I think there is an issue somewhere for a related problem: if logging to stderr, and the stderr stream is blocked (eg piped through `less`), the server will become unavailable when the logging blocks.

**Comment by tbg** [comment]
@knz it's https://github.com/cockroachdb/cockroach/issues/7646, folded into https://github.com/cockroachdb/cockroach/issues/7882
